src/main/kotlin/no/nav/familie/ba/sak/kjerne/personident/MergeIdentService.kt

```diff
@@ -0,0 +1,106 @@
+package no.nav.familie.ba.sak.kjerne.personident
+
+import no.nav.familie.ba.sak.common.Feil
+import no.nav.familie.ba.sak.common.secureLogger
+import no.nav.familie.ba.sak.integrasjoner.pdl.PersonopplysningerService
+import no.nav.familie.ba.sak.integrasjoner.pdl.domene.IdentInformasjon
+import no.nav.familie.ba.sak.integrasjoner.pdl.domene.hentAktivAktørId
+import no.nav.familie.ba.sak.kjerne.behandling.domene.BehandlingRepository
+import no.nav.familie.ba.sak.kjerne.fagsak.FagsakService
+import no.nav.familie.ba.sak.kjerne.grunnlag.personopplysninger.PersongrunnlagService
+import no.nav.familie.ba.sak.task.OpprettTaskService
+import no.nav.familie.ba.sak.task.PatchMergetIdentDto
+import no.nav.familie.kontrakter.felles.PersonIdent
+import no.nav.familie.prosessering.error.RekjørSenereException
+import no.nav.person.pdl.aktor.v2.Type
+import org.springframework.context.annotation.Lazy
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+
+@Service
+class MergeIdentService(
+    private val aktørIdRepository: AktørIdRepository,
+    @Lazy private val fagsakService: FagsakService,
+    private val opprettTaskService: OpprettTaskService,
+    private val persongrunnlagService: PersongrunnlagService,
+    private val personopplysningerService: PersonopplysningerService,
+    private val behandlingRepository: BehandlingRepository,
+) {
+    fun mergeIdentOgRekjørSenere(alleIdenterFraPdl: List<IdentInformasjon>) {
+        val alleHistoriskeAktørIder = alleIdenterFraPdl.filter { it.gruppe == Type.AKTORID.name && it.historisk }.map { it.ident }
+
+        val aktiveAktørerForHistoriskAktørIder =
+            alleHistoriskeAktørIder
+                .mapNotNull { aktørId -> aktørIdRepository.findByAktørIdOrNull(aktørId) }
+                .filter { aktør -> aktør.personidenter.any { personident -> personident.aktiv } }
+
+        if (aktiveAktørerForHistoriskAktørIder.isNotEmpty()) {
```

Comment:
Jeg synes det er greit med `guard cluases` for å unngå nesting. 

Her kan du vel kanskje skirve:

```suggestion
if (aktiveAktørerForHistoriskAktørIder.isEmpty()) {
    return
}
```

Da signaliserer man tydelig at man ikke trenger å tenke noe på koden under om if-en er oppfylt